drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -477,11 +477,13 @@ def do_attempt(self, data, suffix=''):
             misplaced_items.append(self._get_item_definition(int(item_id)))
 
         feedback_msgs = [FeedbackMessage(item['feedback']['incorrect'], None) for item in misplaced_items]
-
+        grade = self._get_raw_earned_if_set()
+        if grade is not None:
+            grade = self.weighted_grade()
```

Comment:
I should just condense all of this into what it is: `_get_weighted_earned_if_set`.